backend/src/packages/chats/chat.service.ts

```diff
@@ -83,8 +96,31 @@ class ChatService implements Service {
     return Promise.resolve(null);
   }
 
-  public delete(): ReturnType<Service['delete']> {
-    return Promise.resolve(true);
+  public async delete({
+    id,
+    userId,
+  }: {
+    id: number;
+    userId: number;
+  }): Promise<boolean> {
+    const chat = await this.findById(id);
+    if (!chat) {
+      throw new ChatError({
+        status: HTTPCode.NOT_FOUND,
+        message: ExceptionMessage.CHAT_NOT_FOUND,
+      });
+    }
+
+    if (chat.members[FIRST_ARRAY_INDEX]?.userId !== userId) {
```

Comment:
what if user is not in first index?

Reply:
1. user can have any position in members array 
2. let's do this in query (query only chats where user is included in)

Reply:
It is not about it. In redux we have this structure. So findById method will return this structure. And we will have this if we have a chat and we will have it, because if no there the error will be thrown earlier.
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/a3144c05-1605-4759-9b07-86cc18f702ec)

Reply:
it's a backend package, not frontend package and as I see here you check if chat related to user by checking if he is the first member, but we need to check if he is included in members

Reply:
we don't even need findById method, we should delete by query and check if the deleted count is 1. Take a look at this pr where userId check also will be added in query https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/387

Reply:
Yes. You can see the response when I try to delete it from frontend. So I always have this first member with userId
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/8312adfc-9732-492b-8b44-648094f8ca6d)

Reply:
Oh, some of your messages are shown only now. So, I will think about it.